test/base/TextTest.ooc

```diff
@@ -91,50 +91,57 @@ TextTest: class extends Fixture {
 			for (i in 1 .. 100)
 				for (j in 1 .. 100)
-					expect(Text new((i * j) toString()) toInt() == i * j)
+					expect(Text new((i * j) toString()) toInt(), is equal to(i * j))
 		})
-		this add("converting to float", func {
-			expect(Text new("1") toFloat() == 1.0f)
-			expect(Text new("-1.0") toFloat() == -1.0f)
-			expect(Text new("-1.") toFloat() == -1.0f)
-			expect(Text new("22.5") toFloat() == 22.5f)
-			expect(Text new("123.763") toFloat() == 123.763f)
-			expect(Text new("1e0") toFloat() == 1.0f)
-			expect(Text new("5E-2") toFloat() == 0.05f)
-			expect(Text new("2E12") toLDouble() == 2.0 * pow(10, 12))
-			expect(Text new("6.5E5") toFloat() == 6.5 * pow(10, 5))
-			expect(Text new("-34.5E-2") toFloat() == -0.345)
+		this add("Convert to Int (base 16)", func {
+			expect(Text new("bad") toInt~inBase(16), is equal to(11 * 16 * 16 + 10 * 16 + 13))
+			expect(Text new("BEEF") toInt(), is equal to(0))
+			expect(Text new("BEEF") toInt~inBase(16), is equal to(48879))
+			expect(Text new("BEEF") toInt~inBase(16), is equal to(Text new("beef") toInt~inBase(16)))
+			expect(Text new("0xff") toInt(), is equal to(255))
+			expect(Text new("0x11") toInt(), is equal to(17))
+			expect(Text new("0xAA") toInt(), is equal to(170))
+			expect(Text new("0xffZZZ") toInt(), is equal to(255))
+		})
+		this add("Convert to Int (other bases)", func {
+			expect(Text new("101") toInt~inBase(2), is equal to(5))
+			expect(Text new("101") toInt~inBase(8), is equal to(8 * 8 + 1))
+			expect(Text new("101") toInt~inBase(7), is equal to(7 * 7 + 1))
+			expect(Text new("654") toInt~inBase(6), is equal to(0))
+			expect(Text new("654") toInt~inBase(7), is equal to(4 + 5 * 7 + 6 * 7 * 7))
+		})
+		this add("Convert to Long and ULong", func {
+			expect(Text new(INT_MAX toString()) toLong(), is equal to(INT_MAX))
+			expect(Text new("0xDEADBEEF") toULong(), is equal to(3735928559))
+			expect(Text new("-9") toULong(), is equal to(0))
+			expect(Text new("-9") toLLong(), is equal to(-9))
+			expect(Text new("-9") toLong(), is equal to(-9))
+		})
+		this add("Convert to Float", func {
+			tolerance := 0.001f
+			expect(Text new("1") toFloat(), is equal to(1.0f) within(tolerance))
+			expect(Text new("-1.0") toFloat(), is equal to(-1.0f) within(tolerance))
+			expect(Text new("-1.") toFloat(), is equal to(-1.0f) within(tolerance))
+			expect(Text new("22.5") toFloat(), is equal to(22.5f) within(tolerance))
+			expect(Text new("123.763") toFloat(), is equal to(123.763f) within(tolerance))
+			expect(Text new("1e0") toFloat(), is equal to(1.0f) within(tolerance))
+			expect(Text new("5E-2") toFloat(), is equal to(0.05f) within(tolerance))
+			expect(Text new("2E12") toLDouble(), is equal to(2.0 * pow(10, 12) as LDouble) within(tolerance as LDouble))
+			expect(Text new("6.5E5") toFloat(), is equal to(6.5f * pow(10, 5) as Float) within(tolerance))
+			expect(Text new("-34.5E-2") toFloat(), is equal to(-0.345f) within(tolerance))
```

Comment:
Since there are so many cases for `Float`, how about putting the ones with `E` in them in their own test function to keep the number of expects down?